lib/src/empty_placeholder.dart

```diff
@@ -0,0 +1,255 @@
+  final String? errorSubtitle;
+  final String? emptyTitle;
+  final String? emptySubtitle;
+  final LoaderName? loadingWidgetName;
+  final Widget? loadingWidget;
+  final Color? loadingColor;
+  final double? loadingSize;
+  final String? animationName;
+  final bool isLoadingOnTop;
+
+  const EmptyListPlaceholder({
+    Key? key,
+    @required this.state,
+    this.onButtonClick,
+    this.placeholderImage,
+    this.darkPlaceholderImage,
+    this.extraPadding,
+    this.buttonText,
+    this.buttonWidth,
+    this.buttonColor,
+    this.buttonTextColor,
+    this.hideButton = false,
+    this.loadingTitle,
+    this.titleTextStyle,
+    this.loadingSubtitle,
+    this.subTitleTextStyle,
+    this.errorTitle,
+    this.errorSubtitle,
+    this.emptyTitle,
+    this.emptySubtitle,
+    this.loadingWidgetName,
+    this.loadingWidget,
+    this.loadingColor,
+    this.loadingSize,
+    this.animationName,
+    this.isLoadingOnTop = false,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    String? title;
+    String? subtitle;
+
+    switch (state) {
+      case ApiState.loading:
+        {
+          title = loadingTitle ?? AppStrings.holdOn;
+          subtitle = loadingSubtitle;
+        }
+        break;
+
+      case ApiState.success:
+        {
+          title = emptyTitle ?? AppStrings.unavailable;
+          subtitle = emptySubtitle;
+        }
+        break;
+
+      case ApiState.error:
+        {
+          title = errorTitle ?? AppStrings.oops;
+          subtitle = errorSubtitle ?? AppStrings.somethingWentWrong;
+        }
+        break;
+
+      // ApiState.idle
+      default:
+        {
+          title = '';
+          subtitle = '';
+        }
+        break;
+    }
+    return SizedBox.expand(
+      child: Container(
+        padding: const EdgeInsets.all(40),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            if (placeholderImage?.isNotEmpty ?? false)
+              Flexible(
+                child: AspectRatio(
+                  aspectRatio: 1,
+                  child: Container(
+                    padding: extraPadding ?? const EdgeInsets.all(0),
+                    child: VectorAsset(
+                      image: context.isDarkTheme
+                          ? darkPlaceholderImage ?? placeholderImage
+                          : placeholderImage,
+                      animationName: animationName,
+                      fit: BoxFit.contain,
+                      width: 10,
+                      height: 10,
+                    ),
+                  ),
+                ),
+              ),
+            isLoadingOnTop
+                ? _changeLoaderPositionToTop(title, subtitle)
+                : _changeLoaderPosition(title, subtitle),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _changeLoaderPosition(String? title, String? subtitle) {
```

Comment:
the function name is not appropriate as it is not changing the position it creating a widget.